libglusterfs/src/iobuf.c

```diff
@@ -678,6 +744,13 @@ iobuf_put(struct iobuf *iobuf)
 
     GF_VALIDATE_OR_GOTO("iobuf", iobuf, out);
 
+    if (iobuf->page_size <= USE_IOBUF_POOL_IF_SIZE_GREATER_THAN) {
```

Comment:
An alternative is to check if `iobuf->iobuf_arena` is NULL.

Reply:
done